akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+alowing the models to be completely detached from each other, such that they can be optimised independently
+as long as the mapping logic is able to convert between them:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#detach-models-adapter
+
+**Additional technique - human readable events using "pass-through" journals:**
+
+The above described technique of detaching the domain model from the data model can be used in interesting ways
+when the journal can be smart about the type of events it receives. For example, let us consider a fictional journal
+plugin implementation that is able to receive ``JSON`` objects, and store them natively in the underlying datastore.
+An example of a datastore which may well implement this pattern is MongoDB, however other databases such as PostgreSQL
+and Cassandra could also do it because of their built-in JSON capabilities.
+
+In this aproach, the :class:`EventAdapter` is used as the marshalling layer: it serializes the events to/from JSON.
+The journal plugin notices that the incoming event type is JSON (for example by performing a ``match`` on the incoming
+event) and
+
+.. note::
+  This technique only applies if the Akka Persistence plugin you are using provides this capability.
+  Check the documentation of your favourite plugin to see if it supports this style of persistence.
+
+  If it doesn't, you may want to skim the `list of existing journal plugins`_, just in case some other plugin
+  for your favourite datastore *does* provide this capability.
+
+.. _list of existing journal plugins: http://akka.io/community/#journal-plugins
+
+Split large event into fine-grained events
+------------------------------------------
+
+**Situation:**
+While refactoring your domain events, you find that one of the events has become too large (coarse-grained)
+and needs to be split up into multiple fine-grained events.
+
+**Solution:**
+Let us consider a situation where an event represents "user details changed". After some time we discover that this
+event is too coarse, and needs to be split into "user name changed" and "user address changed", because somehow
+users keep changing their usernames a lot and we'd like to keep this as a separate event.
+
+The write side change is very simple, we simply persist ``UserNameChanged`` or ``UserAddressChanged`` depending
+on what the user actually intended to change (instead of the composite ``UserDetailsChanged`` that we had in version 1
+of our model).
+
+.. figure:: ../images/persistence-event-adapter-1-n.png
+   :align: center
+
+   The ``EventAdapter`` splits the incoming event into smaller more fine grained events during recovery.
+
+During recovery however, we now need to convert the old ``V1`` model into the ``V2`` representation of the change.
+Depending if the old event contains a name change, we either emit the ``UserNameChanged`` or we don't,
```

Comment:
That is not possible to keep track of without building up state in the event adapter, and that would become very complicated. I think it is reasonable that old events generate both UserNameChanged and AddressChanged no matter what change it actually was.